src/modules/Dashboard/state/CalendarEvents/CalendarEvents.utils.test.ts

```diff
@@ -1,17 +1,1124 @@
+    {
+      ...props,
+      id: expect.any(String),
+      start: getDateFromDateTimeString(dates[0], startTime),
+      end: getDateFromDateTimeString(dates[0], DEFAULT_END_TIME),
+      eventCurrentDate: formatToYearMonthDate(dates[0]),
+      eventSpanAfter: true,
+    },
+    {
+      ...props,
+      id: expect.any(String),
+      start: getDateFromDateTimeString(addDays(dates[0], 1), DEFAULT_START_TIME),
+      end: getDateFromDateTimeString(addDays(dates[0], 1), endTime),
+      eventCurrentDate: formatToYearMonthDate(addDays(dates[0], 1)),
+      eventSpanBefore: true,
+    },
+    {
+      ...props,
+      id: expect.any(String),
+      start: getDateFromDateTimeString(dates[1], startTime),
+      end: getDateFromDateTimeString(dates[1], DEFAULT_END_TIME),
+      eventCurrentDate: formatToYearMonthDate(dates[1]),
+      eventSpanAfter: true,
+    },
+    {
+      ...props,
+      id: expect.any(String),
+      start: getDateFromDateTimeString(addDays(dates[1], 1), DEFAULT_START_TIME),
+      end: getDateFromDateTimeString(addDays(dates[1], 1), endTime),
+      eventCurrentDate: formatToYearMonthDate(addDays(dates[1], 1)),
+      eventSpanBefore: true,
+    },
+  ];
+  const desc1 = 'returns an array with two events from two dates for a non-cross-day event';
+  const desc2 = 'returns an array with four events from two for a cross-day event';
+  const desc3 = 'returns an empty array for empty dates array';
+
+  test.each`
+    dates    | commonProps | startTime    | endTime    | isCrossDayEvent | expected     | description
+    ${dates} | ${props}    | ${startTime} | ${endTime} | ${false}        | ${expected1} | ${desc1}
+    ${dates} | ${props}    | ${startTime} | ${endTime} | ${true}         | ${expected2} | ${desc2}
+    ${[]}    | ${props}    | ${startTime} | ${endTime} | ${false}        | ${[]}        | ${desc3}
+  `('$description', ({ dates, commonProps, startTime, endTime, isCrossDayEvent, expected }) => {
+    const result = getEventsArrayFromDates({
+      dates,
+      commonProps,
+      startTime,
+      endTime,
+      isCrossDayEvent,
+    });
+
+    expect(result).toHaveLength(expected.length);
+    expect(result).toEqual(expected);
+  });
+});
+
+describe('createEvents', () => {
+  const defaultStartDate = '2023-12-06';
+  const defaultEndDate = '2023-12-07';
+  const startTime = '09:25:00';
+  const endTime = '22:30:00';
+  const props1 = {
+    activityOrFlowId: '123',
+    eventId: '1234',
+    activityOrFlowName: 'some name',
+    periodicityType: Periodicity.Once,
+    selectedDate: defaultStartDate,
+    startDate: defaultStartDate,
+    endDate: defaultEndDate,
+    startTime,
+    endTime,
+    isAlwaysAvailable: false,
+    colors: ['red', 'blue'],
+    flowId: 'flow-555',
+    currentYear: 2023,
+    oneTimeCompletion: false,
+    accessBeforeSchedule: true,
+    timerType: TimerType.Timer,
+    timer: 60,
+    notification: null,
+  };
+  const expected1 = [
+    {
+      activityOrFlowId: '123',
```

Comment:
Could you please add a variable with shared properties for `expected1, ..., expected6`